packages/algob/src/lib/tx.ts

```diff
@@ -266,16 +269,64 @@ export async function makeAndSignTx (
   return [txns, signedTxn];
 }
 
+/**
+ * Signs transaction object/s and returns raw signed transaction
+ * @param txn Transaction object/s
+ * @param signer signer
+ */
+export function signTransactionObject (
+  txns: Transaction | Transaction[], signer: wtypes.Signer | wtypes.Signer[]
+): Uint8Array | Uint8Array[] {
+  if (Array.isArray(txns)) {
+    txns = algosdk.assignGroupID(txns);
+    let signedTxn;
+    if (Array.isArray(signer)) {
+      signedTxn = txns.map((txn: Transaction, index: number) => {
+        return signTransaction(txn, signer[index]);
+      });
+    } else {
+      signedTxn = txns.map((txn: Transaction, index: number) => {
+        return signTransaction(txn, signer);
+      });
+    }
+    return signedTxn;
+  } else {
+    if (Array.isArray(signer)) {
+      throw new Error("cannot sign single transaction with multiple signers");
+    }
+    return signTransaction(txns, signer);
+  }
+}
+
 /**
  * Execute single transaction or group of transactions (atomic transaction)
+ * executes `ExecParams` or `Transaction` Object, SDK Transaction object passed to this function
+ * will be signed and sent to network. User can use SDK functions to create transactions.
+ * Note: If passing transaction object a signer/s must be provided.
  * @param deployer Deployer
  * @param execParams transaction parameters or atomic transaction parameters
  * https://github.com/scale-it/algo-builder/blob/docs/docs/guide/execute-transaction.md
  */
 export async function executeTransaction (
   deployer: Deployer,
-  execParams: wtypes.ExecParams | wtypes.ExecParams[]):
-  Promise<ConfirmedTxInfo> {
+  execParams: (wtypes.ExecParams | Transaction) | (wtypes.ExecParams[] | Transaction[]),
+  signer?: wtypes.Signer | wtypes.Signer[]
+): Promise<ConfirmedTxInfo> {
+  if ((Array.isArray(execParams) && wtypes.isTransaction(execParams)) || wtypes.isTransaction(execParams)) {
+    if (signer === undefined) {
+      throw new Error("Signer is not defined");
+    }
+    const signedTxn = signTransactionObject(execParams, signer);
+    const confirmedTx = await deployer.sendAndWait(signedTxn);
+    console.log(confirmedTx);
+    return confirmedTx;
+  }
+
+  // Update type here because we are sure this is not transaction object type
+  execParams = Array.isArray(execParams)
+    ? execParams as wtypes.ExecParams[]
+    : execParams;
```

Comment:
It seems to me we can also consider keeping this as a separate function. Maybe `executeSDKTransaction`? Is there much of an overlap?

Reply:
the idea was to extend `executeTransaction`. So that it is easier for user

Reply:
`executeTransaction(depoloyer, execParams)`

`executeTransaction(depoloyer, sdkTxn, signer)`

They seem different, Ok.

Reply:
What do you mean ?